probe-rs/src/flashing/loader.rs

```diff
@@ -443,9 +443,9 @@ impl FlashLoader {
             .iter()
             // filter for algorithims that contiain adress range
             .filter(|&fa| {
-                fa.flash_properties
```

Comment:
Why did you change this? Is this what makes the tests fail?

A flash algo can span multiple regions (as it was before). As you did now, a region can span multiple flash algorithms (wrong).

Reply:
Ah sorry, I am not used to look at commit messages for such explanations which is kind of bad :/

> Is that not the case?

No, that is indeed not the case :) It just means the Flash algo would be used for any NVMs that could be there IF there was any (can be the case for bigger chips e.g. which then use the same flash algo).

> The new MIMXRT targets have the opposite condition: the NVM region is larger than the flash algorithm address range. The filter as previously written never selects any flash algorithm, even though the single NVM region could accommodate the flash algorithm's range.

What you say here is a valid point ofc. I wonder how we handle this issue best. I wonder if NXP messed up here or if I missunderstand something. Maybe @Tiwalun or @adamgreig  has an idea. I will also have a look at the spec.

Reply:
> It just means the Flash algo would be used for any NVMs that could be there IF there was any (can be the case for bigger chips e.g. which then use the same flash algo).

Ahh thank you! I think I see the distinction.

Would it be safe for us to filter on the _intersection_ of these ranges, instead of requiring that one range is a _subset_ of the other? Changing `contains_range` to `intersects_range` in the original filter means that both our tests pass. Naively, it makes sense that flashing succeeds if we can place the program with the address ranges supported by both NVM and the flashing algorithm(s), but I'm still learning here.

Reply:
The problem with this is that this does not ensure that the entire data block is contained in the memory segment associated with the flash algorithm which has to be the case (at least I think so).

Does anyone of the @probe-rs/core have any input here?